### DISCUSSION TITLE
Get existing controlplane.yaml

### DISCUSSION THREAD

**Discussion by TSTsankov** [siderolabs/talos#9288] [Q&A] [answered]
Pretty new to this, but I've encountered the following situation:

1 - have setup everything on a single raspberry pi (for now) by following the guide
2 - wanted to set a pod acting as a private docker registry - up until encountering an 509 openssl error
2.1 - figured I have to somehow trust the custom CA - i.e. maybe put it in the controlplane.yaml
2.2 - generated the controlplane.yaml, worker.yaml, secrets & talosconfig + uncomented insecureSkipVerify
3 - applied the controlplane.yaml => **broke everything**
4 - reinstalled everything, did the same without changing anything in the controlplane.yaml => still broke everything

So, from what I've gathered - talosctl gen config --with-secrets secrets.yaml, generates some default configuration files, different from the machine config, and without manually adjusting them - the result will be an incorrect configuration

Given that there is _talosctl edit machineconfig_ - is there a straight forward way to get the controlplane.yaml, worker.yaml, secrets & talosconfig from it?

Cheers & any suggestions for learning resources are welcome

**Comment by smira** [comment, marked as answer]
Machine configuration flow while separating and preserving secrets is described here: https://www.talos.dev/v1.7/introduction/prodnotes/#separating-out-secrets

`secrets.yaml` can be recovered from `controlplane.yaml` via `talosctl gen secrets --from-controlplane-config`.

Machine configuration can be downloaded back from the node via `talosctl get mc -o yaml | yq .spec`

**Reply by TSTsankov** [reply]
So the machineconfig.yaml can be used to generate the secrets?
Nonetheless, I think the generated controlplane.yaml (with the secrets.yaml presumably generated from the machineconfig.yaml) will still have properties that will break the machine.

**Reply by smira** [reply]
I don't quite understand your question.

If you follow the docs above, the secrets should be generated first, and then used to generated other machine configuration files. In order to keep things reproducible, you keep around secrets, config patches, and then you can generate machine config as you need.

Make sure you keep `--talos-version=v1.7` set and don't change it for the cluster lifetime (this will keep defaults intact).

**Reply by TSTsankov** [reply]
When you follow this guide, you use `talosctl apply-config --insecure --mode=interactive --nodes <node IP or DNS name>`, which doesn't seem to be generating the initial config files - i.e. no configplane.yaml, no worker.yaml & etc.. 

https://www.talos.dev/v1.7/talos-guides/install/single-board-computers/rpi_generic/

**Reply by smira** [reply]
This is a simplified guide to make you up and running quick and easy.

If you want to move into more of a production setup, use https://www.talos.dev/v1.7/introduction/prodnotes/
